airflow/models/pool.py

```diff
@@ -106,6 +107,8 @@ def slots_stats(
 
         pool_rows: Iterable[Tuple[str, int]] = query.all()
         for (pool_name, total_slots) in pool_rows:
+            if total_slots == -1:
+                total_slots = maxsize
```

Comment:
The one downside to this is that it would then be logged as `9223372036854775807 slots free` etc.

It breaks/is lying about the type, but how about:

```suggestion
                total_slots = float('inf')  # type: ignore
```

Because this has all the properties we want:

- Anything subtracted from it is still infinity
- It is greater than any actual number.

Reply:
Great idea, and existing log end up pretty solid too:
```
[2021-06-04 13:37:28,963] {scheduler_job.py:993} INFO - Figuring out tasks to run in Pool(name=infinite_pool) with inf open slots and 1 task instances ready to be queued
[2021-06-04 13:37:28,963] {scheduler_job.py:1021} INFO - DAG SchedulerJobTest.test_infinite_pool has 0/16 running and queued tasks
[2021-06-04 13:37:28,963] {scheduler_job.py:1086} INFO - Setting the following tasks to queued state:
	<TaskInstance: SchedulerJobTest.test_infinite_pool.dummy 2016-01-01 00:00:00+00:00 [scheduled]>
```